fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2048,23 +2035,56 @@ public int compare(Integer indexA, Integer indexB) {
     }
 
     /**
-     * Returns the specified BundleEntry's path component of the 'url' field.
+     * This method will build a mapping of local identifiers to external identifiers for bundle entries
+     * which specify local identifiers and which have a request method of POST or PUT.
      *
-     * @param entry
-     *            the bundle entry
-     * @return the bundle entry's 'url' field's path component
+     * @param requestBundle
+     *            the bundle containing the requests
+     *            
+     * @return local reference map
      */
-    private static String getUrlPath(Bundle.Entry entry) {
-        String path = null;
-        Bundle.Entry.Request request = entry.getRequest();
-        if (request != null) {
-            if (request.getUrl() != null && request.getUrl().getValue() != null) {
-                FHIRUrlParser requestURL = new FHIRUrlParser(request.getUrl().getValue());
-                path = requestURL.getPath();
+    private Map<String, String> buildLocalRefMap(Bundle requestBundle, Bundle responseBundle) throws Exception {
+        Map<String, String> localRefMap = new HashMap<>();
+
+        for (int entryIndex=0; entryIndex<requestBundle.getEntry().size(); ++entryIndex) {
+            Bundle.Entry requestEntry = requestBundle.getEntry().get(entryIndex);
```

Comment:
use an entryset

Reply:
Used this form of the for loop because I need to get both the requestBundle entry and the corresponding responseBundle entry.